lib/engine-addon.js

```diff
@@ -1034,16 +1034,14 @@ module.exports = {
     options.contentFor = function(type, config) {
       if (type === 'head') {
         let engineConfig = this.engineConfig(config.environment, {});
+        let childrenConfig = this.eachAddonInvoke('contentFor', [type, config]);
+        let content = [];
 
-        let content =
-          '<meta name="' +
-          options.name +
-          '/config/environment" ' +
-          'content="' +
-          escape(JSON.stringify(engineConfig)) +
-          '" />';
+        if (Object.keys(engineConfig).length > 0) {
+          content.push(`<meta name="${options.name}/config/environment" content="${escape(JSON.stringify(engineConfig))}" />`);
+        }
 
-        return content;
+        return content.concat(childrenConfig).join('\n');
```

Comment:
missing tests to cover this new scenario

Reply:
added!

Reply:
@thoov can you also add unit tests in `/node-tests/ultils/engine-addon-test.js` ?